### ISSUE TITLE
[BUG] @marcomasser 's issue with iOS Files app

### ISSUE THREAD

**Issue by michaelstingl** [owncloud/ios-app#1240]
@marcomasser I messed up with the log files and screen recording by mixing issues, users, mail, central forum whatever. Please help us better understand your problem by uploading app debug logs and screen recording from the same time here:
https://cloud.owncloud.com/s/xtYiJ0fDTndof60

@felix-schwarz please access internally here:
https://cloud.owncloud.com/f/6025930

**Comment by marcomasser** [comment]
Don’t worry, these things happen 🙂 @felix-schwarz [said](https://github.com/owncloud/ios-app/issues/1237#issuecomment-1620263241) to continue the discussion on the forum. Which one should I use now? And if this issue here: Should I re-upload the things I uploaded during our email conversation (I used https://cloud.owncloud.com/s/nBZLsdWBtpSEN0m for those)?

**Comment by michaelstingl** [comment]
> Should I re-upload the things I uploaded during our email conversation

Yes, please re-upload to https://cloud.owncloud.com/s/xtYiJ0fDTndof60, and post the name of the log file in forum or here.

**Comment by marcomasser** [comment]
I just uploaded a new screen recording and associated log file:
- ownCloud_06.07.2023__20_59_29.log.txt.zip
- RPReplay_Final1688670028.MP4

**Comment by marcomasser** [comment]
Ping @felix-schwarz ☝️

**Comment by felix-schwarz** [comment]
@marcomasser Thanks for the logs. Based on the findings there, I have put together a new build with more debug output and improved error handling that might already address your issue.

That build is currently going through TestFlight review from Apple. I'll post another comment when it has passed review.

**Comment by felix-schwarz** [comment]
@marcomasser The build (272) has been approved by Apple and is now [available via TestFlight](https://testflight.apple.com/join/MaKmwsqZ). Please give it a try.

**Comment by marcomasser** [comment]
Nice, thanks! I’m on vacation this week without access to that iPad, but I will report back next week!

**Comment by marcomasser** [comment]
I installed the new build, restarted the iPad and recorded a new video, along with a log (log level: debug). It seems like the new build does improve things a bit, but Files.app still gets stuck in a state where it loads for too long. But at least with this build, I saw for the first time that Files.app gets out of the loading state after switching away from the app and back again. So something really did improve!

I uploaded two new files:

- ownCloud_16.07.2023__13_24_38.log.zip
- RPReplay_Final1689506776.MOV

Hope this helps!

**Comment by felix-schwarz** [comment]
@marcomasser Thanks for your help! I'll take a look.

**Comment by felix-schwarz** [comment]
@marcomasser I fixed an issue the FileProvider likely tripped over. Please try the new [TestFlight](https://testflight.apple.com/join/MaKmwsqZ) build 273 that includes the changes.

**Comment by marcomasser** [comment]
Nice, the new build feels much more stable! I tested once again after installing the new build and restarting the iPad. I noticed one hiccup where Files.app showed the loading spinner while navigating into the folder _7_ and I became impatient enough to navigate back but that did kick it out of the hang again (although navigating into some folders was slow, but at least it worked). I uploaded a log and video again:
- ownCloud_21.07.2023__10_16_56.log.zip
- RPReplay_Final1689927518 (trimmed).mov

But this seems to work much better than before already. Also, I tried to reproduce the issue again for a second screen recording (the first recording accidentally showed some patient names, so I trimmed it at the beginning) but everything worked fine there.

Please let me know if there’s anything in the logs that hints at a problem. I told my wife that she could start using her iPad normally again, i.e. that she can edit files on ownCloud directly again (as opposed to editing copies locally on the iPad) and I will report back if I find any issues again. Thank you very much so far!

**Comment by felix-schwarz** [comment]
@marcomasser Thanks once again for the logs and the recording!

I identified a timing issue in your logs that the new [TestFlight](https://testflight.apple.com/join/MaKmwsqZ) build 274 should fix. Please give it a try.

**Comment by marcomasser** [comment]
I just now found time to test the new build and it looks very good! I was not able to reproduce any of hanging issues I saw earlier. Thank you for the fix – good job! 👏 I would consider this issue closed now. If anything comes up again, I’ll let you know.

Just for completeness’ sake, I uploaded a log while I was navigating into the same folders as before:
- ownCloud_28.07.2023__09_20_07.log.txt.zip
